subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java

```diff
@@ -62,6 +63,7 @@
         this.transitiveExclusions = transitiveExclusions;
         this.resolveState = resolveState;
         this.selector = resolveState.getSelector(dependencyState, dependencyState.getModuleIdentifier());
+        this.isTransitive = from.isTransitive() && dependencyMetadata.isTransitive();
```

Comment:
⭕️ I don't see this in the flame graphs, are you sure this is necessary?

Reply:
Yes. This is particularly visible in some graphs where `isTransitive` calls are visible, although they should be transparent, since almost no-op. Basically it happens because the call sites are megamorphic, so the JIT cannot inline calls to the fields directly.